### PR TITLE
Nearby Transit Filter item

### PR DESCRIPTION
**Pull request by TomasSkar**
Nearby transit filter implementation

image

### CODE REVIEW

File: android/transit/ui/src/main/java/com/example/transit/ui/TransitFilter.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+// Todo to ui?
+@ColorInt
+fun String?.toColorInt(@ColorInt fallback: Int = 0xFF_000000.toInt()): Color {
+    var color = this?.toLongOrNull(radix = 16) ?: return Color(fallback)
+    if (length == 6) {
+        // if transparency is not specified, default to opaque
+        color = color or 0x00000000_FF000000
+    }
+    return Color(color.toInt())
+}
```
Comment (TomasSkar) [inline android/transit/ui/src/main/java/com/example/transit/ui/TransitFilter.kt:91]: Will be moved to commonMain -> ui -> Color.kt

Comment (TomasSkar) [reply android/transit/ui/src/main/java/com/example/transit/ui/TransitFilter.kt:95]: https://github.com/trafi/maas-components/pull/64

File: android/ui/src/main/java/com/trafi/ui/components/Filter.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            modifier = Modifier.clickable(onClick = { onItemClick(item) })
+        ) {
+            itemView(item)
+        }
+    }
+}
+
+@Composable
+fun <T> MultiSelectFilter(
+    viewsItems: List<T>,
```
Comment (justasm) [inline android/ui/src/main/java/com/trafi/ui/components/Filter.kt:33]: Perhaps just `items`?

File: android/ui/src/main/res/drawable/navigation_chevron_up.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (justasm) [inline android/ui/src/main/res/drawable/navigation_chevron_up.xml:1]: Are these assets from our Figma whitelabel icon library?

File: android/ui/src/main/java/com/trafi/ui/theme/CornerRadius.kt [hunk 4]
```diff
@@ -18,10 +18,11 @@ object CornerRadius {
...
 @Stable
-internal val Dp.isRound: Boolean
+val Dp.isRound: Boolean
     get() = this == CornerRadius.round
 
 @Immutable
 data class MaasCornerRadius(
-    val buttonRadius: Dp = CornerRadiusScale.Default.ButtonRadius
+    val buttonRadius: Dp = CornerRadiusScale.Default.ButtonRadius,
+    val nearbyTransitFilterItemRadius: Dp = CornerRadiusScale.Default.NearbyTransitFilterItemRadius
```
Comment (justasm) [inline android/ui/src/main/java/com/trafi/ui/theme/CornerRadius.kt:27]: Why should this be part of our public theme API? Unless we have a good reason, I'd suggest hardcoding it in `NearbyTransitFilterItemConstants` for now.

```kotlin
val cornerRadius = CornerRadiusScale.round
```

**Review by justasm — CHANGES_REQUESTED** [review]

File: android/ui/src/main/java/com/trafi/ui/theme/CornerRadius.kt (hunk 4 above)
Comment (Ignelios) [reply android/ui/src/main/java/com/trafi/ui/theme/CornerRadius.kt:27]: Agreed!

File: android/ui/src/main/res/drawable/navigation_chevron_up.xml (hunk 3 above)
Comment (TomasSkar) [reply android/ui/src/main/res/drawable/navigation_chevron_up.xml:1]: Yes

File: android/transit/ui/src/main/java/com/example/transit/ui/TransitFilter.kt [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+import com.trafi.ui.theme.internal.toColor
+import com.trafi.ui.theme.isRound
+
+@Composable
+private val constants
+    get() = NearbyTransitFilterItemConstants(currentTheme)
+
+@Composable
+fun NearbyTransitFilterItem(
+    modifier: Modifier = Modifier,
```
Comment (justasm) [inline android/transit/ui/src/main/java/com/example/transit/ui/TransitFilter.kt:36]: Specify the optional `modifier` parameter as the first *optional* parameter, i.e. after `filterItem` and `isEnabled`

File: android/ui/src/main/java/com/trafi/ui/components/Button.kt [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package com.trafi.ui
+package com.trafi.ui.components
```
[the remark this answers is not in the document]
Comment (justasm) [reply android/ui/src/main/java/com/trafi/ui/components/Button.kt:1]: Seems still unresolved 🤔

File: android/ui/src/main/java/com/trafi/ui/components/Filter.kt [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.lazy.LazyRowFor
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+
+@Composable
+private fun <T> ScrollableTappableView(
+    modifier: Modifier = Modifier,
```
Comment (justasm) [inline android/ui/src/main/java/com/trafi/ui/components/Filter.kt:13]: Ditto, this should be first *optional* parameter

**Review by justasm — APPROVED** [review]

**Pull request merged**